Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CalculateSelfScatteringCorrection.py

```diff
@@ -0,0 +1,113 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.simpleapi import (CalculatePlaczekSelfScattering, ConvertToDistribution, ConvertUnits, CreateWorkspace,
+                              DeleteWorkspace, DiffractionFocussing, Divide, ExtractSpectra, FitIncidentSpectrum,
+                              LoadCalFile, SetSample)
+from mantid.api import (DataProcessorAlgorithm, AlgorithmFactory, WorkspaceProperty)
+from mantid.kernel import Direction
+import numpy as np
+
+
+class CalculateSelfScatteringCorrection(DataProcessorAlgorithm):
```

Comment:
The naming is tricky here as this one is a little but more specific in exactly what it does with the fit first then calculating the Placzek correction. With other areas we've prefixed the name of the area to the algorithm name, e.g Dgs or SANS, to indicate that it's really a utility algorithm for that area.
Maybe something like `TotScatCalculateSelfScattering` although I'm not wedded to that...

Reply:
I've gone with TotScatCalculateSelfScattering, there are no other algorithms that work specifically with polaris or handle total scattering things.